### DISCUSSION TITLE
v2.0.0 Beta 4

### DISCUSSION THREAD

**Discussion by Kovah** [Kovah/LinkAce#856] [Announcements]
<p align="center">
<img src="https://github.com/user-attachments/assets/53a4efdf-e4e5-4982-a1ff-47a4eb213688" alt="LinkAce v2 Beta 4"/>
</p>

### This is the fourth BETA release of LinkAce 2

Changes from v2.0 Beta 3 to Beta 4:

- The import process was overhauled for better processing of large import files. Importing of links is now done in the background.
- Bulk adding for links was added to the API.
- Minor fixes and adjustments for the final release.

**Full Changelog**: https://github.com/Kovah/LinkAce/compare/v2.0.0-beta3...v2.0.0-beta4

<hr /><em>This discussion was created from the release <a href='https://github.com/Kovah/LinkAce/releases/tag/v2.0.0-beta4'>v2.0.0 Beta 4</a>.</em>

---

Please have a look at the [Beta 1 release notes](https://github.com/Kovah/LinkAce/discussions/832) for more details about the Beta phase of LinkAce 2.

To upgrade from Beta 1 to Beta 2, follow the standard upgrade guide for LinkAce.

**Comment by MortRainey** [comment]
I can't add links thrugh the API. I always get "The url field is required" 

`curl --request POST \
--url http://xxxx/api/v2/links \
--header 'accept: application/json' \
--header 'Content-Type: application/json' \
--header 'authorization: Bearer xxxxxx' \
--data '{
"url":"https://duckduckgo.com",
"title":"DuckDuckGo"
"is_private":true,
"check_disabled":true
}'`

**Reply by Kovah** [reply]
The following request was successful:
```bash
curl --request POST \
  --url https://demo-v2.linkace.org/api/v2/links \
  --header 'accept: application/json' \
  --header 'authorization: Bearer s9anum7ODLsxbMQCr4WnYSahCEZViFEB' \
  --header 'content-type: application/json' \
  --data '{ "url": "https://duckduckgoo.com", "title": "DuckDuck", "visibility": 1, "check_disabled": false }'
```
Please note that the is_private field is no longer supported, see [the new v2 docs](https://api-docs.linkace.org/v2#tag/links/operation/post-api-v2-links).
I am not sure what exactly the problem is with your request, as I don't see any significant difference beside the is_private field.

**Reply by MortRainey** [reply]
I don't know why, but it's working again. I have no clue what went wrong. 

Thanks for the hint with the "is_private" field. I'm using the beta and it's working very well until now.

**Reply by Kovah** [reply]
Great to hear and thanks for trying the beta!
